src/mem/periph/downrupt.rs

```diff
@@ -1,15 +1,38 @@
-use serde::{Deserialize, Serialize};
-use zmq;
+use crate::utils::generate_yaagc_packet;
+
+use crossbeam_channel::{Sender, Receiver, unbounded};
+use std::io::Write;
+use std::net::TcpListener;
 
 pub struct DownruptPeriph {
-    state: [u16; 200],
-    current_idx: usize,
-    socket: zmq::Socket,
+    tx: Sender<[u8; 4]>
 }
 
-#[derive(Serialize, Deserialize, Debug)]
-struct DownruptMessage {
-    state: Vec<u16>,
+fn downrupt_thread(rx: Receiver<[u8; 4]>) {
+    // accept connections and process them serially
+    let listener = TcpListener::bind("127.0.0.1:19800").unwrap();
+    for stream in listener.incoming() {
+        match stream {
+            Ok(mut xa) => {
+                loop {
+                    let msg = match rx.recv() {
+                        Ok(x) => x,
+                        _ => {
+                            break;
+                        }
+                    };
+
+                    match xa.write(&msg) {
```

Comment:
You can lose data here. `write` is fallible and also doesn’t guarantee complete transmission of bytes.

Reply:
Good catch. I know this and the dsky follow the same approach so they probably have the same issue. Initial thought is to capture the results and break out of the loop if any errors occur. 

Curious on the guarantee of the complete transmission of bytes. My assumption was either it sends all or errors out at any point in between 0 and n. My assumption is the yaTelemetry will handle the parsing if channels are disconnected and reconnected.

Reply:
You are looking for `write_all` which behaves that way.

Reply:
I will create a new issue to update my `write` to `write_all`. Thanks for letting me know.